packages/amplify-migration-tests/src/migration-helpers-v10/init.ts

```diff
@@ -0,0 +1,94 @@
+
+import { addCircleCITags, getCLIPath, nspawn as spawn } from '@aws-amplify/amplify-e2e-core';
+import { EOL } from 'os';
+
+const defaultSettings = {
+    name: EOL,
+    // eslint-disable-next-line spellcheck/spell-checker
+    envName: 'integtest',
+    editor: EOL,
+    appType: EOL,
+    framework: EOL,
+    srcDir: EOL,
+    distDir: EOL,
+    buildCmd: EOL,
+    startCmd: EOL,
+    useProfile: EOL,
+    profileName: EOL,
+    region: process.env.CLI_REGION,
+    local: false,
+    disableAmplifyAppCreation: true,
+    disableCIDetection: false,
+    providerConfig: undefined,
+    permissionsBoundaryArn: undefined,
+};
+  
+export function initJSProjectWithProfileV10(cwd: string, settings?: Partial<typeof defaultSettings>): Promise<void> {
+    const s = { ...defaultSettings, ...settings };
+    let env;
+  
+    if (s.disableAmplifyAppCreation === true) {
+      env = {
+        CLI_DEV_INTERNAL_DISABLE_AMPLIFY_APP_CREATION: '1',
+      };
+    }
+  
+    addCircleCITags(cwd);
+  
+    const cliArgs = ['init'];
+    const providerConfigSpecified = !!s.providerConfig && typeof s.providerConfig === 'object';
+    if (providerConfigSpecified) {
+      cliArgs.push('--providers', JSON.stringify(s.providerConfig));
+    }
+  
+    if (s.permissionsBoundaryArn) {
+      cliArgs.push('--permissions-boundary', s.permissionsBoundaryArn);
+    }
+  
+    if (s?.name?.length > 20) console.warn('Project names should not be longer than 20 characters. This may cause tests to break.');
```

Comment:
Should we throw an error here to enforce we don't run into this instead?

Reply:
I copied this method as a snapshot from the current v10 e2e-core utils; i can modify but seems out of scope for this pr